### PR TITLE
[WIP] chore: upgrade Rust crates

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -145,6 +145,12 @@ version = "0.12.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "3441f0f7b02788e948e47f457ca01f1d7e6d92c693bc132c22b087d3141c03ff"
 
+[[package]]
+name = "base64"
+version = "0.13.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "904dfeac50f3cdaba28fc6f57fdcddb75f49ed61346676a78c4ffe55877802fd"
+
```
Comment (ry) [inline Cargo.lock:153]: unfortunate that we're introducing a 3rd version of base64...

File: Cargo.lock [hunk 2]
```diff
@@ -260,15 +266,23 @@ version = "0.1.10"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "4785bdd1c96b2a846b2bd7cc02e86b6b3dbf14e7e53446c4f54c92a361040822"
 
+[[package]]
+name = "cfg-if"
+version = "1.0.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "baf1de4339761588bc0619e3cbc0120ee582ebb74b53b4efbf79117bd2da40fd"
```
Comment (ry) [inline Cargo.lock:273]: and a second version of cfg-if ...

File: Cargo.lock [hunk 3]
```diff
@@ -1578,7 +1598,16 @@ version = "0.4.23"
...
+]
+
+[[package]]
+name = "pin-project"
+version = "1.0.2"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9ccc2237c2c489783abd8c4c80e5450fc0e98644555b1364da68cc29aa151ca7"
+dependencies = [
+ "pin-project-internal 1.0.2",
 ]
```
Comment (ry) [inline Cargo.lock:1611]: and a second version of pin-project (plus their new pin-project-internal crate... sigh)

**Review by ry — APPROVED** [review]
LGTM

**Comment by bartlomieju** [comment]
@ry I'm not done with update, most of duplicates should be gone once I upgrade `rusty_v8` and `swc` related crates

**Comment by bartlomieju** [comment]
I'll open new PR once new version of `rusty_v8` is released.

**Pull request closed without merging**